core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/Scheduler.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.scheduler
+
+class Scheduler {}
+
+object Scheduler {
+
+  def main(args: Array[String]): Unit = {
```

Comment:
why is this an executable/main? I thought the scheduler will implement some existing interface.

Reply:
Sorry, I don't get your point.

The scheduler is a dedicated component which will act as a "scheduler" to schedule container creation and a queue to hold activations instead of Kafka.
https://cwiki.apache.org/confluence/display/OPENWHISK/System+Architecture